### PR TITLE
2.1.6 backport commits

### PR DESCRIPTION
**Pull request by kit-ty-kate**
Backports:
- [x] https://github.com/ocaml/opam/pull/5869 (partially)
- [x] https://github.com/ocaml/opam/pull/5893
- [x] https://github.com/ocaml/opam/pull/5065
- [x] https://github.com/ocaml/opam/pull/5769

Fixes:
- All the issues listed in https://github.com/ocaml/opam/pull/5400
  - required to avoid angry macOS users when the current proposal in https://github.com/ocaml/opam-repository/issues/23789 is applied
- Both issues reported in https://github.com/ocaml/opam/issues/5770
  - allow to use an up-to-date dune to build opam (due to a wrong parsing from `promote-install-files`)
  - vendor dune >= 3.7.0 (fixes PPC support)
- improved the detection of installed system packages (depexts) on NetBSD, OpenBSD and Gentoo
- debug logs showing on BSDs and macOS
  
~In its current form the PR is only a proposal. In particular given the amount of untested changes coming from the bump to lang dune 2.0, it might be worth instead detect the version of dune required for the `--promote-install-files` flag (>= 2.0) instead of bumping the lang version.~

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
> In particular given the amount of untested changes coming from the bump to lang dune 2.0, it might be worth instead detect the version of dune required for the `--promote-install-files` flag (>= 2.0) instead of bumping the lang version.

Actually this whole change is unwelcome for 2.1... :( as dune 2.0 drops support for OCaml < 4.06 and we need to support OCaml >= 4.02. I'll go back to the drawing board

**Comment by kit-ty-kate** [comment]
Changing the grep pattern worked just fine, yay

**Comment by kit-ty-kate** [comment]
As discussed during the [dev meeting](https://github.com/ocaml/opam/wiki/2024-Developer-Meetings#2024-03-11), this should wait for one week after the release of `2.2.0~beta2` to give a bit more time for the new usage of `git apply` to be more wildly tested

**Comment by kit-ty-kate** [comment]
CI is all green (except for ocaml-benchmarks as usual which can't be turned off for this branch)

**Review by dra27 — APPROVED** [review]
LGTM! (over to @rjbou in the morning)

File: master_changes.md [hunk 1]
```diff
@@ -54,6 +56,16 @@ New option/command/subcommand are prefixed with ◈.
   *
 
 ## Build
+  * Bump src_exts and fix build compat with Dune 2.9.0 [#4752 @dra27]
+  * Upgrade to dose3 >= 6.1 and vendor dose3 7.0.0 [#4760 @kit-ty-kate]
+  * Change minimum required OCaml to 4.03.0 [#4770 @dra27]
+  * Change minimum required Dune to 2.0 [#4770 @dra27]
+  * Do not check for cppo in the configure script (not used directly anymore since #5498) [#5794 @kit-ty-kate]
+  * Upgrade vendored cmdliner to 1.2.0 [#5797 @kit-ty-kate]
+  * Add winsymlinks:native to the CYGWIN environment variable when installing a package on Windows [#5793 @kit-ty-kate - fix #5782]
```
[the remark this answers is not in the document]
Comment (rjbou) [reply master_changes.md]: Plus missing entry for 5893

**Review by rjbou — APPROVED** [review]
Updated the PR with CHANGES and bum version

**Comment by kit-ty-kate** [comment]
Thanks a lot!

As usual ocaml-benchmarks is the only CI job failing, ignoring.

**Pull request merged**